File: NavDrawerDemo/NavDrawerDemo/app/src/main/java/com/example/navdrawer/MainActivity.java
package com.example.navdrawer;

import android.content.Intent;
import android.os.Bundle;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import android.view.Gravity;
import android.view.MenuItem;
import android.view.View;

import androidx.navigation.NavController;
import androidx.navigation.Navigation;
import androidx.navigation.ui.AppBarConfiguration;
import androidx.navigation.ui.NavigationUI;

import com.google.android.material.navigation.NavigationView;

import androidx.drawerlayout.widget.DrawerLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.Menu;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private AppBarConfiguration mAppBarConfiguration;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //call the toolbar
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        //find the floating action button
        FloatingActionButton fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                //launches a simple snackbar, we can replace this with whatever we want, an intent for example
                Snackbar.make(view, "I am a snackbar, popup", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();
            }
        });

        //this finds the accompanying drawerLayout
        DrawerLayout drawer = findViewById(R.id.drawer_layout);

        //this simply finds the navigation view.
        NavigationView navigationView = findViewById(R.id.nav_view);

        // Passing each menu ID to the app bar configuration
        mAppBarConfiguration = new AppBarConfiguration.Builder(
                R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow,
                R.id.nav_tools, R.id.nav_share, R.id.nav_send, R.id.nav_cake)
                .setDrawerLayout(drawer)
                .build();

        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration);
        NavigationUI.setupWithNavController(navigationView, navController);

        /*
*/
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);

        MenuItem mySettings = menu.findItem(R.id.action_settings);
        MenuItem mySave = menu.findItem(R.id.action_save);

        //listen for a selection on settings.
        mySettings.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
            @Override
            public boolean onMenuItemClick(MenuItem item) {
                Intent intent = new Intent(MainActivity.this, PartyTime.class);
                startActivity(intent);
                return false;
            }
        });

        //listen for a selection on save.
        mySave.setOnMenuItemClickListener(new MenuItem.OnMenuItemClickListener() {
            @Override
            public boolean onMenuItemClick(MenuItem item) {

                Toast fancy = new Toast(MainActivity.this);
                fancy.setDuration(Toast.LENGTH_SHORT);

                int offset = fancy.getYOffset();
                fancy.setGravity(Gravity.TOP, 0, offset);

                fancy.setView(getLayoutInflater().inflate(R.layout.toast_demo, null));
                fancy.show();

                return false;
            }
        });

        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onSupportNavigateUp() {
        NavController navController = Navigation.findNavController(this, R.id.nav_host_fragment);
        return NavigationUI.navigateUp(navController, mAppBarConfiguration)
                || super.onSupportNavigateUp();
    }

    public void randomIntent(MenuItem item) {
        Intent intent = new Intent(this, PartyTime.class);
        startActivity(intent);
    }
}

File: Kotlin/NavDrawerDemo/app/src/main/java/com/example/navdrawer/MainActivity.kt
package com.example.navdrawer

import android.content.Intent
import android.os.Bundle
import android.view.Gravity
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.drawerlayout.widget.DrawerLayout
import androidx.navigation.Navigation
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.NavigationUI
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.google.android.material.navigation.NavigationView
import com.google.android.material.snackbar.Snackbar

class MainActivity : AppCompatActivity() {
    private var mAppBarConfiguration: AppBarConfiguration? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //call the toolbar
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        //find the floating action button
        val fab = findViewById<FloatingActionButton>(R.id.fab)
        fab.setOnClickListener { view -> //launches a simple snackbar, we can replace this with whatever we want, an intent for example
            Snackbar.make(view, "I am a snackbar, popup", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }

        //this finds the accompanying drawerLayout
        val drawer = findViewById<DrawerLayout>(R.id.drawer_layout)

        //this simply finds the navigation view.
        val navigationView = findViewById<NavigationView>(R.id.nav_view)

        // Passing each menu ID to the app bar configuration
        mAppBarConfiguration = AppBarConfiguration.Builder(
                R.id.nav_home, R.id.nav_gallery, R.id.nav_slideshow,
                R.id.nav_tools, R.id.nav_share, R.id.nav_send, R.id.nav_cake)
                .setDrawerLayout(drawer)
                .build()
        val navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        NavigationUI.setupActionBarWithNavController(this, navController, mAppBarConfiguration!!)
        NavigationUI.setupWithNavController(navigationView, navController)

        /*
*/
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        val mySettings = menu.findItem(R.id.action_settings)
        val mySave = menu.findItem(R.id.action_save)

        //listen for a selection on settings.
        mySettings.setOnMenuItemClickListener {
            val intent = Intent(this@MainActivity, PartyTime::class.java)
            startActivity(intent)
            false
        }

        //listen for a selection on save.
        mySave.setOnMenuItemClickListener {
            val fancy = Toast(this@MainActivity)
            fancy.duration = Toast.LENGTH_SHORT
            val offset = fancy.yOffset
            fancy.setGravity(Gravity.TOP, 0, offset)
            fancy.view = layoutInflater.inflate(R.layout.toast_demo, null)
            fancy.show()
            false
        }
        return super.onCreateOptionsMenu(menu)
    }

    override fun onSupportNavigateUp(): Boolean {
        val navController = Navigation.findNavController(this, R.id.nav_host_fragment)
        return (NavigationUI.navigateUp(navController, mAppBarConfiguration!!)
                || super.onSupportNavigateUp())
    }

    fun randomIntent(item: MenuItem?) {
        val intent = Intent(this, PartyTime::class.java)
        startActivity(intent)
    }
}